feat: git branches nesting

FezVrasta opened the pull request:
<!--- IMPORTANT: If this PR addresses multiple unrelated issues, it will be closed until separated. -->

### Description

Implements Git branches grouping using the first slash as prefix separator. This allows for easier navigation on large repositories that prefix branches (for example: `feat/some-feature`)

This PR builds on top of https://github.com/CodeEditApp/CodeEdit/pull/1570

### Related Issues

<!--- REQUIRED: Tag all related issues (e.g. * #123) -->
<!--- If this PR resolves the issue please specify (e.g. * closes #123) -->
<!--- If this PR addresses multiple issues, these issues must be related to one other -->

* #1569

### Checklist

<!--- Add things that are not yet implemented above -->

- [x] I read and understood the [contributing guide](https://github.com/CodeEditApp/CodeEdit/blob/main/CONTRIBUTING.md) as well as the [code of conduct](https://github.com/CodeEditApp/CodeEdit/blob/main/CODE_OF_CONDUCT.md)
- [x] The issues this PR addresses are related to each other
- [x] My changes generate no new warnings
- [x] My code builds and runs on my machine
- [x] My changes are all related to the related issue above
- [x] I documented my code

### Screenshots

![CleanShot 2024-01-22 at 4  07 39@2x](https://github.com/CodeEditApp/CodeEdit/assets/5382443/68ab0683-e00f-4300-905d-e16fe09aed8a)

austincondiff reviewed (approved):
Looks good to me! Please fix the SwiftLint errors before we can merge.

FezVrasta commented:
What do you mean? What website?

austincondiff commented:
@Angelk90 I think @FezVrasta has provided sufficient details for this PR.

activcoding commented:
@FezVrasta thanks for the PR! Got some questions:

 1: Can we use the popover? I really like its style:
<img width="79" alt="Screenshot 2024-01-24 at 1 28 04 PM" src="https://github.com/CodeEditApp/CodeEdit/assets/83090745/a544e0b5-ba3a-414a-ac1a-a1e6118ef537">

2. How about more symbols for branches and a coloured checkmark for the current branch? 
<img width="117" alt="Screenshot 2024-01-24 at 1 26 21 PM" src="https://github.com/CodeEditApp/CodeEdit/assets/83090745/62804068-c32f-47e3-9642-2a8d7f7bc6cf">

3. Any chance to add more details to each branch section?
<img width="289" alt="Screenshot 2024-01-24 at 1 30 14 PM" src="https://github.com/CodeEditApp/CodeEdit/assets/83090745/0b331cc3-7514-4862-99b8-dbd71b3de13e">

FezVrasta commented:
I fixed the linter and also removed the branch prefix from the nested dropdown entries.
![CleanShot 2024-01-24 at 1  47 58@2x](https://github.com/CodeEditApp/CodeEdit/assets/5382443/0fee19f1-5f8f-4d33-a5bb-6dc401e84faa)

FezVrasta commented:
> @FezVrasta: Great, what happens if there is only one branch and it is the selected one.
> 
> In the future I would propose to change the branch icon, I would propose like that of xcode.

I haven't changed that behavior so it will work like it does today.

austincondiff commented:
This looks fantastic @FezVrasta! I would agree, we have a "branch" symbol in CodeEditSymbols that you might use instead.

FezVrasta commented:
Done

![CleanShot 2024-01-25 at 3  47 14@2x](https://github.com/CodeEditApp/CodeEdit/assets/5382443/d6b4c424-4765-4ec5-a9fe-d9ae3d981de3)

austincondiff reviewed (approved):
Great work @FezVrasta! 👏

FezVrasta commented:
![CleanShot 2024-01-25 at 7  31 48@2x](https://github.com/CodeEditApp/CodeEdit/assets/5382443/ff8fae55-89de-4a6e-a371-485b15270710)

austincondiff commented:
Yeah, let's merge this PR. We are scope creeping quite a bit here imo. Looks good enough to merge.

FezVrasta commented:
> @FezVrasta thanks for the PR! Got some questions:
> 
> 1: Can we use the popover? I really like its style: <img alt="Screenshot 2024-01-24 at 1 28 04 PM" width="79" src="https://private-user-images.githubusercontent.com/83090745/299312279-a544e0b5-ba3a-414a-ac1a-a1e6118ef537.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYyMDg4ODUsIm5iZiI6MTcwNjIwODU4NSwicGF0aCI6Ii84MzA5MDc0NS8yOTkzMTIyNzktYTU0NGUwYjUtYmEzYS00MTRhLWFjMWEtYTFlNjExOGVmNTM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDE4NDk0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzNTJhZTYzZjM1OTdiNTczMzEyODg0YzZlMWJiNjZlOGZjMTk0YWQxOWNkMzNmYWM1MTM0ZWQ1NzNkZDhlNjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IwUGQHowMqFIvUaxqcfnZmw4tf4NU1m2MNL2OXoVne4">
> 
> 2. How about more symbols for branches and a coloured checkmark for the current branch?
> 
> <img alt="Screenshot 2024-01-24 at 1 26 21 PM" width="117" src="https://private-user-images.githubusercontent.com/83090745/299311900-62804068-c32f-47e3-9642-2a8d7f7bc6cf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYyMDg4ODUsIm5iZiI6MTcwNjIwODU4NSwicGF0aCI6Ii84MzA5MDc0NS8yOTkzMTE5MDAtNjI4MDQwNjgtYzMyZi00N2UzLTk2NDItMmE4ZDdmN2JjNmNmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDE4NDk0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2ZWY5ZmVmOTYxMjM5YzQzZTliZTBkYmMzZmU1OTEwNmYxNTQzODVhZGYzZWMxMmRmNGM3NzQ0YTIzNjBmN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fx0esGnxo1ykcW7o4AuPnHt5saskjVMRRF7jHcfLgwo">
> 3. Any chance to add more details to each branch section?
> 
> <img alt="Screenshot 2024-01-24 at 1 30 14 PM" width="289" src="https://private-user-images.githubusercontent.com/83090745/299312857-0b331cc3-7514-4862-99b8-dbd71b3de13e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYyMDg4ODUsIm5iZiI6MTcwNjIwODU4NSwicGF0aCI6Ii84MzA5MDc0NS8yOTkzMTI4NTctMGIzMzFjYzMtNzUxNC00ODYyLTk5YjgtZGJkNzFiM2RlMTNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDE4NDk0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhNzNhYzg3ZmIyNDVlNTZiNzhiMjQxYWUzZGI4ZTMwOTU1ZTI1NTQ1MzJlZjNlN2MzMTkwMTc3NzNjYmEwZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Waozhp66RmrjtdU0NX6crur_8rd-IsCF9DTM0lgRE_s">

I used the Menu component because it has built-in keyboard support, we can definitely use the Popover but I think we'll have to build keyboard support from scratch?

activcoding commented:
> > @FezVrasta thanks for the PR! Got some questions:
> > 1: Can we use the popover? I really like its style: <img alt="Screenshot 2024-01-24 at 1 28 04 PM" width="79" src="https://private-user-images.githubusercontent.com/83090745/299312279-a544e0b5-ba3a-414a-ac1a-a1e6118ef537.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYyMDg4ODUsIm5iZiI6MTcwNjIwODU4NSwicGF0aCI6Ii84MzA5MDc0NS8yOTkzMTIyNzktYTU0NGUwYjUtYmEzYS00MTRhLWFjMWEtYTFlNjExOGVmNTM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDE4NDk0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzNTJhZTYzZjM1OTdiNTczMzEyODg0YzZlMWJiNjZlOGZjMTk0YWQxOWNkMzNmYWM1MTM0ZWQ1NzNkZDhlNjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IwUGQHowMqFIvUaxqcfnZmw4tf4NU1m2MNL2OXoVne4">
> > 
> > 2. How about more symbols for branches and a coloured checkmark for the current branch?
> > 
> > <img alt="Screenshot 2024-01-24 at 1 26 21 PM" width="117" src="https://private-user-images.githubusercontent.com/83090745/299311900-62804068-c32f-47e3-9642-2a8d7f7bc6cf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYyMDg4ODUsIm5iZiI6MTcwNjIwODU4NSwicGF0aCI6Ii84MzA5MDc0NS8yOTkzMTE5MDAtNjI4MDQwNjgtYzMyZi00N2UzLTk2NDItMmE4ZDdmN2JjNmNmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDE4NDk0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2ZWY5ZmVmOTYxMjM5YzQzZTliZTBkYmMzZmU1OTEwNmYxNTQzODVhZGYzZWMxMmRmNGM3NzQ0YTIzNjBmN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fx0esGnxo1ykcW7o4AuPnHt5saskjVMRRF7jHcfLgwo">
> > 3. Any chance to add more details to each branch section?
> > <img alt="Screenshot 2024-01-24 at 1 30 14 PM" width="289" src="https://private-user-images.githubusercontent.com/83090745/299312857-0b331cc3-7514-4862-99b8-dbd71b3de13e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYyMDg4ODUsIm5iZiI6MTcwNjIwODU4NSwicGF0aCI6Ii84MzA5MDc0NS8yOTkzMTI4NTctMGIzMzFjYzMtNzUxNC00ODYyLTk5YjgtZGJkNzFiM2RlMTNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDE4NDk0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhNzNhYzg3ZmIyNDVlNTZiNzhiMjQxYWUzZGI4ZTMwOTU1ZTI1NTQ1MzJlZjNlN2MzMTkwMTc3NzNjYmEwZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Waozhp66RmrjtdU0NX6crur_8rd-IsCF9DTM0lgRE_s">
> 
> I used the Menu component because it has built-in keyboard support, we can definitely use the Popover but I think we'll have to build keyboard support from scratch?

Ok, great choice going with the Menu component. Your work looks fantastic—let's get this merged! If @austincondiff adds me as a reviewer, I'd be more than happy to go through your changes.

activcoding commented:
> @activcoding : If anything, can you try to add these details yourself?

I don't like your tone here; it feels a bit demanding. Nevertheless, I've got a lot on my plate with a few PRs in progress. Once those are sorted, I'll definitely take a look at the issue.

0xWDG reviewed (approved):
Nice work!
